app/src/main/java/com/eveningoutpost/dexdrip/Models/TransmitterData.java

```diff
@@ -50,77 +50,86 @@
-            //this is a dexbridge packet.  Process accordingly.
-            Log.i(TAG, "create Processing a Dexbridge packet");
-            ByteBuffer txData = ByteBuffer.allocate(len);
-            txData.order(ByteOrder.LITTLE_ENDIAN);
-            txData.put(buffer, 0, len);
-            transmitterData.raw_data = txData.getInt(2);
-            transmitterData.filtered_data = txData.getInt(6);
-            //  bitwise and with 0xff (1111....1) to avoid that the byte is treated as signed.
-            transmitterData.sensor_battery_level = txData.get(10) & 0xff;
-            if (buffer[0] == 0x15) {
-                Log.i(TAG, "create Processing a Dexbridge packet includes delay information");
-                transmitterData.timestamp = timestamp - txData.getInt(16);
-            } else {
-                transmitterData.timestamp = timestamp;
-            }
-            Log.i(TAG, "Created transmitterData record with Raw value of " + transmitterData.raw_data + " and Filtered value of " + transmitterData.filtered_data + " at " + timestamp + " with timestamp " + transmitterData.timestamp);
-        } else { //this is NOT a dexbridge packet.  Process accordingly.
-            Log.i(TAG, "create Processing a BTWixel or IPWixel packet");
-            StringBuilder data_string = new StringBuilder();
-            for (int i = 0; i < len; ++i) { data_string.append((char) buffer[i]); }
-            final String[] data = data_string.toString().split("\\s+");
-
-            if (data.length > 1) {
-                transmitterData.sensor_battery_level = Integer.parseInt(data[1]);
-                if (data.length > 2) {
-                    try {
-                        Home.setPreferencesInt("bridge_battery", Integer.parseInt(data[2]));
-                        if (Home.get_master()) {
-                            GcmActivity.sendBridgeBattery(Home.getPreferencesInt("bridge_battery", -1)); }
-                        CheckBridgeBattery.checkBridgeBattery();
-                    } catch (Exception e) {
-                        Log.e(TAG, "Got exception processing classic wixel or limitter battery value: " + e.toString());
-                    }
-                    if (data.length > 3) {
-                        if ((DexCollectionType.getDexCollectionType() == DexCollectionType.LimiTTer)
-                                && (!Home.getPreferencesBooleanDefaultFalse("use_transmiter_pl_bluetooth"))) {
-                            try {
-                                // reported sensor age in minutes
-                                final Integer sensorAge = Integer.parseInt(data[3]);
-                                if ((sensorAge > 0) && (sensorAge < 200000))
-                                    Home.setPreferencesInt("nfc_sensor_age", sensorAge);
-                            } catch (Exception e) {
-                                Log.e(TAG, "Got exception processing field 4 in classic limitter protocol: " + e);
+        try {
```

Comment:
yes i wrapped the whole section in try/catch to avoid parsing of additional messages from the blueReader